Mathlib/LinearAlgebra/Matrix/Ideal.lean

```diff
@@ -113,141 +114,193 @@ end Ideal
 
 /-! ### Two-sided ideals in a matrix ring -/
 
+namespace RingCon
+variable {R n : Type*}
+
+section NonUnitalNonAssocSemiring
+variable [NonUnitalNonAssocSemiring R] [Fintype n]
+variable (n)
+
+/-- The ring congruence of matrices with entries related by `c`. -/
+def matricesOver (c : RingCon R) : RingCon (Matrix n n R) where
+  r M N := ∀ i j, c (M i j) (N i j)
+  iseqv.refl _ := fun _ _ => c.refl _
+  iseqv.symm h := fun _ _ => c.symm <| h _ _
+  iseqv.trans h₁ h₂ := fun _ _ => c.trans (h₁ _ _) (h₂ _ _)
+  add' h₁ h₂ := fun _ _ => c.add (h₁ _ _) (h₂ _ _)
+  mul' h₁ h₂ := fun _ _ => c.finset_sum _ fun _ _ => c.mul (h₁ _ _) (h₂ _ _)
```

Comment:
```suggestion
  iseqv.refl _ _ _ := c.refl _
  iseqv.symm h _ _ := c.symm <| h _ _
  iseqv.trans h₁ h₂ _ _ := c.trans (h₁ _ _) (h₂ _ _)
  add' h₁ h₂ _ _ := c.add (h₁ _ _) (h₂ _ _)
  mul' h₁ h₂ _ _ := c.finset_sum _ fun _ _ ↦ c.mul (h₁ _ _) (h₂ _ _)
```
ditto for `ofMatricesOver`.
And can you switch to `↦` throughout?

Reply:
This was deliberate, to distinguish binders that are part of the type from binders which are induced by the chosen relation. Do to think this is an anti pattern?

Either way, I'll change the arrow styles.

Reply:
It makes me think there's some elaboration problem that prevent you from getting rid of the `fun` (maybe there was). I think other people may golf it if you leave it this way.

Reply:
Done.